### PR TITLE
Adding Azure Spring Cloud task

### PR DESCRIPTION
**Pull request by yevster**
**Task name**: Azure Spring Cloud

**Description**: Deploys to and manages deployments to Azure Spring Cloud

**Documentation changes required:** Y (New documentation)

**Added unit tests:** Y

**Attached related issue:** (Internal)

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/AzureSpringCloudV0/operations/AzureResourceFilterUtility.ts
Comment (ajinkya599) [inline Tasks/AzureSpringCloudV0/operations/AzureResourceFilterUtility.ts:18]: Move the string to resources?

File: Tasks/AzureSpringCloudV0/operations/AzureResourceFilterUtility.ts
Comment (ajinkya599) [inline Tasks/AzureSpringCloudV0/operations/AzureResourceFilterUtility.ts:9]: As a good practise, use `const` for objects that are not expected to be modified, `let` for objects that may get modified. Prefer `let` over `var`.

File: Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-cloud.ts
Comment (ajinkya599) [inline Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-cloud.ts:248]: Remove extra newlines here and other places.

File: Tasks/AzureSpringCloudV0/tsconfig.json
Comment (yevster) [reply Tasks/AzureSpringCloudV0/tsconfig.json:8]: Yeah, these were added as workarounds to build errors

File: Tasks/AzureSpringCloudV0/task.json
Comment (yevster) [reply Tasks/AzureSpringCloudV0/task.json:20]: Will fix in next push.

File: Tasks/AzureSpringCloudV0/task.json
Comment (yevster) [reply Tasks/AzureSpringCloudV0/task.json:53]: I'm following [the precedent of the App Service Manage task](https://github.com/microsoft/azure-pipelines-tasks/blob/b908bffea681a03d1b8925e299d2a2b794bc035e/Tasks/AzureAppServiceManageV0/task.json#L53). I don't mind deviating from it, if that's your preference.

File: Tasks/AzureSpringCloudV0/task.json
Comment (yevster) [reply Tasks/AzureSpringCloudV0/task.json:106]: Actually, yes! Initially, we anticipated having four actions. However, if anyone ever adds another action, they might have to scratch their head why is suddenly broken. My gut instinct is to make the logic more resilient, even at the cost of slightly greater complexity. You can make the final call.

File: Tasks/AzureSpringCloudV0/operations/AzureResourceFilterUtility.ts
Comment (yevster) [reply Tasks/AzureSpringCloudV0/operations/AzureResourceFilterUtility.ts:9]: Fixed in a places where it was straightforward.

File: Tasks/AzureSpringCloudV0/operations/AzureResourceFilterUtility.ts
Comment (yevster) [reply Tasks/AzureSpringCloudV0/operations/AzureResourceFilterUtility.ts:18]: Done, although in theory, this should never be possible (#FamousLastWords)

File: Tasks/AzureSpringCloudV0/task.json
Comment (ajinkya599) [reply Tasks/AzureSpringCloudV0/task.json]: Possibility is that App Service task was authored before YAML support, so it doesn't matter much as to what is saved in the background, as what's visible in the UI is the friendly name.
But, with YAML support in pipelines, my preference is to not use spaces. Will leave it to you to make the final call.

File: Tasks/AzureSpringCloudV0/task.json
Comment (ajinkya599) [reply Tasks/AzureSpringCloudV0/task.json]: Good to start with `0.185.0`

File: Tasks/AzureSpringCloudV0/Tests/SetProductionUseStagingSucceeds.ts
Comment (ajinkya599) [inline Tasks/AzureSpringCloudV0/Tests/SetProductionUseStagingSucceeds.ts:32]: Remove stdout and stderr from all the tests. I don't see any other tasks doing it. I believe the reason is that this will cause the CI logs (where all tasks are built and tested) to explode.

File: Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-cloud.ts
Comment (ajinkya599) [inline Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-cloud.ts:83]: This parsing can be error prone. (Complicated scenarios like - escaped quotes inside quotes)
Let's discuss this once.

File: Tasks/AzureSpringCloudV0/task.json
Comment (yevster) [reply Tasks/AzureSpringCloudV0/task.json]: I've already exhausted 5 in hotfixes my local organization. :D

File: Tasks/AzureSpringCloudV0/task.json
Comment (yevster) [reply Tasks/AzureSpringCloudV0/task.json]: I would err on the side of consistency and of being done. 😄
As ineloquent as long literals are, they do not add architectural complexity.

**Comment by ajinkya599** [comment]
Add the task to [make-options](https://github.com/microsoft/azure-pipelines-tasks/blob/master/make-options.json) and [CODEOWNERS](https://github.com/microsoft/azure-pipelines-tasks/blob/master/.github/CODEOWNERS)

File: Tasks/AzureSpringCloudV0/azurespringclouddeployment.ts
Comment (yevster) [reply Tasks/AzureSpringCloudV0/azurespringclouddeployment.ts:29]: Done

**Review by ajinkya599 — APPROVED** [review]

**Pull request merged**
